lib/src/Uncertainty/Algorithm/Sensitivity/openturns/HSICEstimatorImplementation.hxx

```diff
@@ -0,0 +1,179 @@
+
+//                                               -*- C++ -*-
+/**
+ * @brief HSICEstimatorImplementation implements the HSIC sensivity index
+ *
+ *  Copyright 2005-2021 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef OPENTURNS_HSICESTIMATORIMPLEMENTATION_HXX
+#define OPENTURNS_HSICESTIMATORIMPLEMENTATION_HXX
+
+#include "openturns/PersistentObject.hxx"
+#include "openturns/Point.hxx"
+#include "openturns/Sample.hxx"
+#include "openturns/Matrix.hxx"
+#include "openturns/Graph.hxx"
+#include "openturns/PersistentCollection.hxx"
+#include "openturns/ResourceMap.hxx"
+#include "openturns/CovarianceModel.hxx"
+#include "openturns/SymbolicFunction.hxx"
+#include "openturns/Function.hxx"
+#include "openturns/HSICStat.hxx"
+#include "openturns/RandomGenerator.hxx"
+BEGIN_NAMESPACE_OPENTURNS
+
+/**
+ * @class HSICEstimatorImplementation
+ *
+ */
+class OT_API HSICEstimatorImplementation
+  : public PersistentObject
```

Comment:
all your objects inherit from PersistentObject but they lack the save/load methods
that is mandatory to be able to serialize the results in xml

Reply:
OK. However, I do not understand how to save/load the `CovarianceModelCollection` argument `covarianceList_`. It produces an error. Is there a different mechanism for collections?

Reply:
yes, it should be a PersistentCollection<CovarianceModel>

Reply:
Thank you, the name is quite telling.